app/services/dashboard/data.rb

```diff
@@ -23,11 +23,14 @@ def latest_events
     con = ActiveRecord::Base.connection
     #workaround to get the inner query working
     ids = hosts.to_sql.sub(/SELECT.*FROM/, "SELECT #{Host.quoted_table_name}.#{con.quote_column_name('id')} FROM")
+    #sql optimization - instead of comparing dates, check just the id
+    oldest_report_id = ConfigReport.search_for('reported > "7 days ago"').last.try(:id)
```

Comment:
I first thought that would be a good change as well, but `.fetch` is no method of `nil`, so in case `.last` returns nil it'll raise an error, `.try` won't, but will return nil.